events/source/equeue.c

```diff
@@ -275,14 +282,29 @@ static int equeue_enqueue(equeue_t *q, struct equeue_event *e, unsigned tick)
 
 static struct equeue_event *equeue_unqueue(equeue_t *q, int id)
 {
-    // decode event from unique id and check that the local id matches
-    struct equeue_event *e = (struct equeue_event *)
-                             &q->buffer[id & ((1 << q->npw2) - 1)];
-
-    equeue_mutex_lock(&q->queuelock);
-    if (e->id != id >> q->npw2) {
-        equeue_mutex_unlock(&q->queuelock);
-        return 0;
+    struct equeue_event *e = 0;
+    if (IS_USER_ALLOCATED_EVENT_ID(id)) {
+        equeue_mutex_lock(&q->queuelock);
```

Comment:
Could pull this lock outside the if, just to minimise code

Reply:
I certainly agree with the general principle, but not convinced putting a single `if (id & 1)` inside a critical section is enough extra overhead to be worth increasing code size to avoid.

The critical section would be better served by eliminating the list search by removing the use of IDs for static events, as suggested previously.